src/discordcr/mappings/guild.cr

```diff
@@ -59,10 +59,21 @@ module Discord
 
   struct GuildMember
     # :nodoc:
-    def initialize(payload : Gateway::GuildMemberAddPayload | GuildMember, roles : Array(UInt64)? = nil)
+    def initialize(payload : Gateway::GuildMemberAddPayload | GuildMember, roles : Array(UInt64), nick : String?)
+      # This redundant assignment is necessary because Crystal would otherwise
+      # erroneously assume @roles can be nil.
```

Comment:
Please make an issue with this and add a TODO linking to the crystal issue before comitting this.

Reply:
IIRC this is actually fixed in crystal now.